src/people.cpp

```diff
@@ -372,7 +372,7 @@ void Guests::AddFree(Guest *g)
  */
 Guest *Guests::GetFree()
 {
-	bool b = this->FindNextFreeGuest();
+	[[maybe_unused]] bool b = this->FindNextFreeGuest();
```

Comment:
lol :) 

Maybe also use this for the `#ifndef NDEBUG` things? Let the compiler figure out if it's useful.

This solution may have the dis-advantage of getting old assignments in the code that are never used due to code using it has been removed in the past.